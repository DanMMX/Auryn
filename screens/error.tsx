import * as React from 'react';
import { View, StyleSheet, Text } from 'react-native';

interface ErrorProps {
  text: string;
}

export const Error: React.FunctionComponent<ErrorProps> = ({ text }) => (
  <View style={styles.container}>
    <Text style={styles.text}>{text}</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    borderColor: 'black',
    borderWidth: 3,
    backgroundColor: '#143672',
    alignItems: 'center',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  text: {
    fontSize: 36,
    color: 'white',
  },
});
