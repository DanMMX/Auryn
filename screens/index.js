import Splash   from './splash.youi'
import Lander   from './lander.youi'
import PDP      from './pdp.youi'
import Player   from './video.youi'

export {
  Splash,
  Lander,
  PDP,
  Player,
}
