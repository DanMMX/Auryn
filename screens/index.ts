/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import Splash from './splash';
import Lander from './lander';
import PDP from './pdp';
import Search from './search';
import Profile from './profile';
import Video from './video';

export {
  Splash,
  Lander,
  PDP,
  Search,
  Profile,
  Video,
};
