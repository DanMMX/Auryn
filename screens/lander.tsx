/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import * as React from 'react';
import { Composition, ViewRef, ScrollRef, ButtonRef, FocusManager, BackHandler } from '@youi/react-native-youi';
import { View, NativeEventSubscription } from 'react-native';
import { Timeline, ToggleGroup, List } from '../components';
import {
  withNavigationFocus,
  NavigationActions,
  NavigationScreenProps,
  NavigationEventSubscription,
} from 'react-navigation';
import { connect, DispatchProp } from 'react-redux';
import { tmdb } from '../actions';
import { Asset } from '../adapters/asset';
import { Config } from '../config';
import { TmdbActionTypes } from '../typings/tmdbReduxTypes';
import { AurynAppState } from '../reducers';
import { ListItemFocusEvent, ListItemPressEvent } from '../components/listitem';
import { ToggleButtonPress } from '../components/toggleButton';

interface LanderProps extends NavigationScreenProps, DispatchProp<TmdbActionTypes> {
    isFocused: boolean;
    tv: Asset[];
    discover: Asset[];
    movies: Asset[];
  };

interface LanderState {
  currentListIndex: number;
}

class Lander extends React.Component<LanderProps, LanderState> {
  state = { currentListIndex: 0 };

  lists: React.RefObject<List>[] = Array(4).fill(React.createRef<List>());

  lastFocusItem = React.createRef<ButtonRef>();

  lastFocusNavItem = React.createRef<ButtonRef>();

  navButtonNames: string[] = ['Discover', 'Movies', 'Shows', 'Live'];

  focusListener!: NavigationEventSubscription;

  blurListener!: NavigationEventSubscription;

  backHandlerListener!: NativeEventSubscription;

  outTimeline = React.createRef<Timeline>();

  navInTimeline = React.createRef<Timeline>();

  inTimeline = React.createRef<Timeline>();

  menuButtons = React.createRef<ToggleGroup>();

  searchButton = React.createRef<ButtonRef>();

  profileButton = React.createRef<ButtonRef>();

  scroller = React.createRef<ScrollRef>();

  componentDidMount() {
    this.focusListener = this.props.navigation.addListener('didFocus', () => {
      this.backHandlerListener = BackHandler.addEventListener('hardwareBackPress', this.navigateBack);

      if (this.lastFocusNavItem && this.lastFocusNavItem.current) {
        FocusManager.enableFocus(this.lastFocusNavItem.current);
        FocusManager.focus(this.lastFocusNavItem.current);
      } else if (this.lastFocusItem && this.lastFocusItem.current) {
        FocusManager.enableFocus(this.lastFocusItem.current);
        FocusManager.focus(this.lastFocusItem.current);
      }

      if (this.inTimeline.current && this.navInTimeline.current) {
        this.navInTimeline.current.play();
        this.inTimeline.current.play();
      }
    });
    this.blurListener = this.props.navigation.addListener('didBlur', () => this.backHandlerListener.remove());

    if (this.menuButtons.current) FocusManager.focus(this.menuButtons.current.getButtonRef(0).current);
  }

  navigateBack = () => {
    if (this.menuButtons.current)
      FocusManager.focus(this.menuButtons.current.getButtonRef(this.state.currentListIndex).current);
    return true;
  };

  componentWillUnmount() {
    this.focusListener.remove();
    this.blurListener.remove();
    this.backHandlerListener.remove();
  }

  navigateToScreen = async (screen: string) => {
    const navigateAction = NavigationActions.navigate({
      routeName: screen,
    });

    if (screen === 'Search' && this.searchButton.current) this.lastFocusNavItem = this.searchButton;
    else if (screen === 'Profile' && this.profileButton.current) this.lastFocusNavItem = this.profileButton;

    if (this.outTimeline.current) await this.outTimeline.current.play();
    this.props.navigation.dispatch(navigateAction);
  };

  scrollToViewByIndex: ToggleButtonPress = index => {
    if (!Config.isRoku) {
      if (this.menuButtons.current) {
        for (let i = 0; i < this.lists.length; i++)
        FocusManager.setNextFocus(this.menuButtons.current.getButtonRef(i).current, this.lists[index].current, 'down');
      }

      if (this.menuButtons.current
        && this.searchButton.current
        && this.profileButton.current
        && this.lists[index].current
      ) {
        FocusManager.setNextFocus(this.searchButton.current, this.lists[index].current, 'down');
        FocusManager.setNextFocus(this.profileButton.current, this.lists[index].current, 'down');
        FocusManager.setNextFocus(this.lists[index].current, this.menuButtons.current.getButtonRef(index).current, 'up');
      }
    }

    this.setState({ currentListIndex: index });
    if (this.scroller.current) {
      this.scroller.current.scrollTo({
        x: 0,
        y: (index * 900) + 1, // eslint-disable-line no-extra-parens
        animated: true,
      });
    }
  };

  // eslint-disable-next-line max-params
  onFocusItem: ListItemFocusEvent = (id, type, ref, shouldChangeFocus) => {
    this.props.dispatch(tmdb.prefetchDetails(id, type));
    this.lastFocusItem = ref;

    if (shouldChangeFocus === false || Config.isRoku || !ref.current) return;

    if (this.menuButtons.current) {
      FocusManager.setNextFocus(ref.current, this.menuButtons.current.getButtonRef(this.state.currentListIndex).current, 'up');
      for (let index = 0; index < this.lists.length; index++)
        FocusManager.setNextFocus(this.menuButtons.current.getButtonRef(index).current, ref.current, 'down');
    }

    if (this.searchButton.current && this.profileButton.current) {
      FocusManager.setNextFocus(this.searchButton.current, ref.current, 'down');
      FocusManager.setNextFocus(this.profileButton.current, ref.current, 'down');
    }
  };

  onPressItem: ListItemPressEvent = async (id, type, ref) => {
    this.lastFocusItem = ref;
    this.lastFocusNavItem = null;
    const navigateAction = NavigationActions.navigate({
      routeName: 'PDP',
      params: {
        id,
        type,
      },
    });
    this.props.dispatch(tmdb.getDetailsByIdAndType(id, type));
    if (this.outTimeline.current) await this.outTimeline.current.play();
    this.props.navigation.dispatch(navigateAction);
  };

  // eslint-disable-next-line max-lines-per-function
  render() {
    const { isFocused, tv, movies, discover } = this.props;
    const { currentListIndex } = this.state;

    // null list is used for Roku
    const nullList = (
      <Composition source="Auryn_Container-NullList">
        <List name="NullList" type="None" data={[]} focusable={false} />
      </Composition>
    );

    const lists = [
      <Composition source="Auryn_Container-Discover" key="discover">
        <List
          name="Discover"
          type="Discover"
          data={discover}
          ref={this.lists[0]}
          focusable={isFocused && currentListIndex === 0}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
      <Composition source="Auryn_Container-Movies" key="movies">
        <List
          name="Movies"
          type="Movies"
          data={movies}
          ref={this.lists[1]}
          focusable={isFocused && currentListIndex === 1}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
      <Composition source="Auryn_Container-Shows" key="shows">
        <List
          name="Shows"
          type="Shows"
          data={tv}
          ref={this.lists[2]}
          focusable={isFocused && currentListIndex === 2}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
      <Composition source="Auryn_Container-Live" key="live">
        <List
          name="Live"
          type="Live"
          data={movies.slice(0, 2)}
          ref={this.lists[3]}
          focusable={isFocused && currentListIndex === 3}
          onFocusItem={this.onFocusItem}
          onPressItem={this.onPressItem}
        />
      </Composition>,
    ];
    return (
      <Composition source="Auryn_Lander">
        <ToggleGroup
          focusable={isFocused}
          prefix="Btn-Nav-"
          names={this.navButtonNames}
          onPressItem={this.scrollToViewByIndex}
          ref={this.menuButtons}
        />
        <ButtonRef
          name="Btn-Nav-Search"
          focusable={isFocused}
          ref={this.searchButton}
          onPress={() => this.navigateToScreen('Search')}
        />
        <ButtonRef
          name="Btn-Nav-Profile"
          focusable={isFocused}
          ref={this.profileButton}
          onPress={() => this.navigateToScreen('Profile')}
        />
        <Timeline name="LanderIn" ref={this.inTimeline} />
        <Timeline name="LanderOut" ref={this.outTimeline} />
        <ScrollRef
          name="Stack"
          ref={this.scroller}
          scrollEnabled={false}
          horizontal={false}
          focusable={false}
        >
          <View>{Config.isRoku ? [lists[currentListIndex], nullList] : lists}</View>
        </ScrollRef>

        <ViewRef name="Nav">
          <Timeline name="In" ref={this.navInTimeline} />
          <Timeline name="Out" />
        </ViewRef>

        <ViewRef name="Nav-Logo">
          <Timeline name="Loop" loop={true} />
        </ViewRef>
      </Composition>
    );
  }
}

const mapStateToProps = (store: AurynAppState) => ({
  discover: store.tmdbReducer.discover.data,
  movies: store.tmdbReducer.movies.data,
  tv: store.tmdbReducer.tv.data,
});

export default withNavigationFocus(connect(mapStateToProps)(Lander));
export { Lander as LanderTest };
