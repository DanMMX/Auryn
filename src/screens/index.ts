/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export { Splash }   from './splash';
export { Lander }   from './lander';
export { Pdp }      from './pdp';
export { Search }   from './search';
export { Profile }  from './profile';
export { VideoScreen as Video }    from './video';
