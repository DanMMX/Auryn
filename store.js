import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';
import promise from 'redux-promise-middleware';
import thunk from 'redux-thunk';

import reducer from './reducers';

const middleware = applyMiddleware(thunk, logger, promise());

export default createStore(reducer, middleware);
