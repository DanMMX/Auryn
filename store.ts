/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { applyMiddleware, createStore } from 'redux';
import promise from 'redux-promise-middleware';
import createDebounce from 'redux-debounced';
import thunk from 'redux-thunk';
import { rootReducer } from './reducers';

const middleware = applyMiddleware(createDebounce(), thunk, promise);

export default createStore(rootReducer, middleware);
