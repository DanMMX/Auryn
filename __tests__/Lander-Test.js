/**
 * Copyright (c) You i Labs Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { LanderTest } from '../screens/lander';
import { Provider } from 'react-redux';
import store from '../store';

import renderer from 'react-test-renderer';

const data = {
  discover: [],
  movies: [],
  tv: [],
  lander: jest.fn(),
};

test('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}>
    <LanderTest navigation={global.navigation} dispatch={jest.fn()} {...data}/>
  </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
