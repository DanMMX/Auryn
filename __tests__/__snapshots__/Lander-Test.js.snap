// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Composition
  onTimelinesLoaded={[Function]}
  source="Auryn_Lander"
>
  <YiPushButtonView
    name="Btn-Nav-Discover"
    onFocus={[Function]}
    onLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      direction="forward"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={true}
    />
  </YiPushButtonView>
  <YiPushButtonView
    name="Btn-Nav-Movies"
    onFocus={[Function]}
    onLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      direction="reverse"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
  </YiPushButtonView>
  <YiPushButtonView
    name="Btn-Nav-Shows"
    onFocus={[Function]}
    onLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      direction="reverse"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
  </YiPushButtonView>
  <YiPushButtonView
    name="Btn-Nav-Live"
    onFocus={[Function]}
    onLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      direction="reverse"
      loop={false}
      name="Toggle-On"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
  </YiPushButtonView>
  <YiPushButtonView
    name="Btn-Nav-Search"
    onLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  />
  <YiPushButtonView
    name="Btn-Nav-Profile"
    onLoad={[Function]}
    onPress={[Function]}
    onTimelinesLoaded={[Function]}
  />
  <TimelineRef
    direction="forward"
    loop={false}
    name="LanderIn"
    onAborted={[Function]}
    onCompleted={[Function]}
    onCompletedForward={[Function]}
    onCompletedReverse={[Function]}
    onLoad={[Function]}
    onPaused={[Function]}
    onPausedForward={[Function]}
    onPausedReverse={[Function]}
    onPlay={[Function]}
    onPlayForward={[Function]}
    onPlayReverse={[Function]}
    onStarted={[Function]}
    onStartedForward={[Function]}
    onStartedReverse={[Function]}
    playOnLoad={false}
  />
  <TimelineRef
    direction="forward"
    loop={false}
    name="LanderOut"
    onAborted={[Function]}
    onCompleted={[Function]}
    onCompletedForward={[Function]}
    onCompletedReverse={[Function]}
    onLoad={[Function]}
    onPaused={[Function]}
    onPausedForward={[Function]}
    onPausedReverse={[Function]}
    onPlay={[Function]}
    onPlayForward={[Function]}
    onPlayReverse={[Function]}
    onStarted={[Function]}
    onStartedForward={[Function]}
    onStartedReverse={[Function]}
    playOnLoad={false}
  />
  <ListRef
    DEPRECATED_sendUpdatedChildFrames={false}
    alwaysBounceHorizontal={false}
    alwaysBounceVertical={true}
    focusable={false}
    horizontal={false}
    name="Stack"
    onContentSizeChange={null}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminationRequest={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollShouldSetResponder={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    scrollBarThumbImage={null}
    scrollEnabled={false}
    sendMomentumEvents={false}
    style={
      Array [
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "overflow": "scroll",
        },
        undefined,
      ]
    }
  >
    <ScrollContentData
      collapsable={false}
      style={
        Array [
          false,
          undefined,
        ]
      }
    >
      <View>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Discover"
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Discover"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminationRequest={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onScrollShouldSetResponder={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onTimelinesLoaded={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="Discover"
            onLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Movies"
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Movies"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminationRequest={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onScrollShouldSetResponder={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onTimelinesLoaded={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="Movies"
            onLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Shows"
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Shows"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminationRequest={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onScrollShouldSetResponder={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onTimelinesLoaded={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="Shows"
            onLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
        <Composition
          onTimelinesLoaded={[Function]}
          source="Auryn_Container-Live"
        >
          <ListRef
            DEPRECATED_sendUpdatedChildFrames={false}
            alwaysBounceHorizontal={true}
            alwaysBounceVertical={false}
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            getItemLayout={[Function]}
            horizontal={true}
            initialNumToRender={2}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            name="Live"
            numColumns={1}
            onContentSizeChange={null}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminationRequest={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            onScrollShouldSetResponder={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            onTimelinesLoaded={[Function]}
            onTouchCancel={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollBarThumbImage={null}
            scrollEnabled={true}
            scrollEventThrottle={50}
            sendMomentumEvents={true}
            stickyHeaderIndices={Array []}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "overflow": "scroll",
                },
                undefined,
              ]
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <ScrollContentData
              collapsable={false}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </ListRef>
          <ViewRef
            name="Live"
            onLoad={[Function]}
            onTimelinesLoaded={[Function]}
          />
        </Composition>
      </View>
    </ScrollContentData>
  </ListRef>
  <ViewRef
    name="Nav"
    onLoad={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      direction="forward"
      loop={false}
      name="In"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
    <TimelineRef
      direction="forward"
      loop={false}
      name="Out"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
  </ViewRef>
  <ViewRef
    name="Nav-Logo"
    onLoad={[Function]}
    onTimelinesLoaded={[Function]}
  >
    <TimelineRef
      direction="forward"
      loop={true}
      name="Loop"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
      playOnLoad={false}
    />
  </ViewRef>
</Composition>
`;
